smart_aquarium/lib/ui/SensorsPage.dart

```diff
@@ -0,0 +1,123 @@
+import 'dart:convert';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:smart_aquarium/ui/MeasurementsPage.dart';
+
+class SensorsPage extends StatefulWidget {
+  final int aquariumId;
+    @override
+  _SensorsPageState createState() => _SensorsPageState();
+
+  SensorsPage(this.aquariumId);
+}
+  class _SensorsPageState extends State<SensorsPage> {
+    List<dynamic> validUserSensors = [];
+
+     @override
+  void initState() {
+    super.initState();
+    _fetchSensors(); 
+  }
+
+Future<List<dynamic>> _fetchSensors() async {
+
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    int aquariumId = prefs.getInt('aquarium_id') ?? 0;
+    
+      final response = await http.get(Uri.parse('http://localhost:6868/api/v1/userSensor'));
+      if (response.statusCode == 200) {
+        final List<dynamic> userSensors = jsonDecode(response.body);
+      validUserSensors = userSensors.where((sensor) => sensor['aquarium']['aquariumId'] == aquariumId).toList();
```

Comment:
Maybe you could store field names like "aquarium", "aquariumId" or "sensorTypeId" in enum (https://dart.dev/language/enums) or global consts accesible in all of the files (or at least ineach file declared in the beggining)

Reply:
What's wrong with lists?